_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -349,6 +356,35 @@ public static function register_endpoints() {
 		) );
 	}
 
+	public static function get_plans( $request ) {
+		$data      = get_transient( 'jetpack_plans' );
+
+		/**
+		 * Filter to turn off caching of Jetpack plans
+		 *
+		 * @since 5.9.0
+		 *
+		 * @param bool true Whether to cache Jetpack plans locally
+		 */
+		$use_cache = apply_filters( 'cache_jetpack_plans', true );
+
+		if ( $data === false ) {
+			$path = '/plans';
+			$ip   = $_SERVER['HTTP_X_FORWARDED_FOR']; // used to calculate which currency to show
+			if ( empty( $ip ) ) {
+				// no ip list to forward, so create one:
+				$ip = $_SERVER['HTTP_CLIENT_IP'];
+			}
+			$data = Jetpack_Client::wpcom_json_api_request_as_blog( $path, '2', array( 'headers' => array( 'X-Forwarded-For' => $ip ) ), null, 'wpcom' );
+
+			if ( $use_cache ) {
```

Comment:
Should we make this more specific, just in case someone decides to use the filter incorrectly and uses the `false` string instead of a boolean for example?